src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -72,10 +73,31 @@ float64_t CGaussianARDKernel::compute(int32_t idx_a, int32_t idx_b)
 	return CMath::exp(-result);
 }
 
+SGVector<float64_t> CGaussianARDKernel::get_parameter_gradient_diagonal(
+		const TParameter* param, index_t index)
+{
+	REQUIRE(param, "Param not set\n");
+	REQUIRE(lhs && rhs, "Features not set!\n");
+
+	if (!strcmp(param->m_name, "weights") || !strcmp(param->m_name, "width"))
+	{
+		int32_t length=CMath::min(num_lhs, num_rhs);
+		SGVector<float64_t> res(length);
+		res.zero();
```

Comment:
yes since GaussianARD kernel is
`exp(\frac {({\bf x}-{\bf y})^T \Lambda^T \Lambda ({\bf x}-{\bf y})} {w})

for diagonal part of gradients, note that x==y, which implies the kernel value is const (1 in this case)